sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1070,6 +1107,7 @@ public Instant getWatermark() {
 
     @Override
     public CheckpointMark getCheckpointMark() {
+      reportBacklog();
```

Comment:
I have not read about Metrics. Is this the recommended place to update backlog metrics?